stac_fastapi/opensearch/stac_fastapi/opensearch/database_logic.py

```diff
@@ -665,6 +665,142 @@ async def execute_search(
+                }
+            },
+            "datetime_min": {"min": {"field": "properties.datetime"}},
+            "datetime_max": {"max": {"field": "properties.datetime"}},
+            "grid_code_frequency": {
+                "terms": {
+                    "field": "properties.grid:code",
+                    "missing": "none",
+                    "size": 10000,
+                }
+            },
+            "sun_elevation_frequency": {
+                "histogram": {"field": "properties.view:sun_elevation", "interval": 5}
+            },
+            "sun_azimuth_frequency": {
+                "histogram": {"field": "properties.view:sun_azimuth", "interval": 5}
+            },
+            "off_nadir_frequency": {
+                "histogram": {"field": "properties.view:off_nadir", "interval": 5}
+            },
+        }
+
+        search_body: Dict[str, Any] = {}
+        query = search.query.to_dict() if search.query else None
+        if query:
+            search_body["query"] = query
+
+        # include all aggregations specified
+        # this will ignore aggregations with the wrong names
+        search_body["aggregations"] = {
+            k: v for k, v in agg_2_es.items() if k in aggregations
+        }
+
+        # centroid
+        if "centroid_geohash_grid_frequency" in aggregations:
+            search_body["aggregations"]["centroid_geohash_grid_frequency"] = {
+                "geohash_grid": {
+                    "field": "properties.proj:centroid",
+                    "precision": centroid_geohash_grid_precision,
+                }
+            }
+
+        if "centroid_geohex_grid_frequency" in aggregations:
+            search_body["aggregations"]["centroid_geohex_grid_frequency"] = {
+                "geohex_grid": {
+                    "field": "properties.proj:centroid",
+                    "precision": centroid_geohex_grid_precision,
+                }
+            }
+
+        if "centroid_geotile_grid_frequency" in aggregations:
+            search_body["aggregations"]["centroid_geotile_grid_frequency"] = {
+                "geotile_grid": {
+                    "field": "properties.proj:centroid",
+                    "precision": centroid_geotile_grid_precision,
+                }
+            }
+
+        # geometry
+        if "geometry_geohash_grid_frequency" in aggregations:
+            search_body["aggregations"]["geometry_geohash_grid_frequency"] = {
+                "geohash_grid": {
+                    "field": "geometry",
+                    "precision": geometry_geohash_grid_precision,
+                }
+            }
+
+        # if "geometry_geohex_grid_frequency" in aggregations:
+        #     search_body["aggregations"]["geometry_geohex_grid_frequency"] = {
+        #         "geohex_grid": {
+        #             "field": "geometry",
+        #             "precision": geometry_geohex_grid_precision,
+        #         }
+        #     }
```

Comment:
Hi. Is this meant to be commented out?

Reply:
Yeah, It is a placeholder for when OS/ES add geohex aggregation for geo_shape geometries

Reply:
It is a bit confusing to have it in the code, though. I do not know when that feature is coming to OS/ES, so I'll remove the commented out code.

Reply:
You can leave the code in if you want - just add a comment explaining why it's commented out

Reply:
Okay. I don't see it on any scheduled release for opensearch. So I'll leave it out

Reply:
it's not AFAIK -- maybe file another GH issue and pop the code that was removed in there, so it's available in future.